### ISSUE TITLE
Ambiguous calls after upgrading to 2.0.0

### ISSUE THREAD

**Issue by KeterSCP** [atata-framework/atata#662]
After upgrading to 2.0.0 we got a lot of errors in our tests in places where we have both usings of `Atata` and `System.Linq` namespaces.
Calling methods like `First`, `FirstOrDefault`, `Select`, etc. on `ControlList` gives an error:
```
[CS0121] The call is ambiguous between the following methods or properties: 'System.Linq.Enumerable.First<TSource>(System.Collections.Generic.IEnumerable<TSource>, System.Func<TSource, bool>)' and 'Atata.IEnumerableProviderExtensions.First<TSource, TOwner>(Atata.IEnumerableProvider<TSource, TOwner>, System.Linq.Expressions.Expression<System.Func<TSource, bool>>)'
```

**Comment by KeterSCP** [comment]
We are using .NET6 and C#10

**Comment by KeterSCP** [comment]
Simple repro:

```c#
using System.Linq;
using Atata;

namespace AtataDemo;

using _ = AtataDemo;

public class AtataDemo : Page<_>
{
    public void DemoMethod<T>() where T : PageObject<T>
    {
        ControlList<Control<AtataDemo>, AtataDemo> controls = default;

        var result = controls.Where(x => x is null);
    }
}
```

**Comment by KeterSCP** [comment]
> @KeterSCP, I have just released v2.0.1 with a fix of this issue. You can find details of fix in #663. Please let me know whether it works for you.

I can confirm that errors no longer appear. Thanks, @YevgeniyShunevych for the quick fix!

**Comment by YevgeniyShunevych** [comment]
Great! You are welcome. Thanks for posting this issue.
